worlds/rabi_ribi/__init__.py

```diff
@@ -200,7 +204,15 @@ def create_items(self) -> None:
                 continue
             self.multiworld.itempool.append(item)
 
-        junk = self.total_locations - len(base_item_list)
+        max_egg_locations_in_pool = self.total_locations - len(base_item_list)
+        total_egg_count = min(max_egg_locations_in_pool, self.options.max_number_of_easter_eggs.value)
+        self.required_egg_count = max(math.floor(total_egg_count * (self.options.percentage_of_easter_eggs.value / 100.0)), 1)
+        filler_egg_count = total_egg_count - self.required_egg_count
+
+        self.multiworld.itempool.extend([self.create_item(ItemName.easter_egg) for _ in range(self.required_egg_count)])
+        self.multiworld.itempool.extend([self.create_item(ItemName.easter_egg, ItemClassification.useful) for _ in range(filler_egg_count)])
```

Comment:
Nit: Does AP only consider the progression eggs to be required for logic? iirc, there's no logic around this, so which eggs AP expects you to have to beat the game doesn't actually correspond with the item classification.

I've found this somewhat confusing in other games as a player, should we just mark them all as progression?

Don't really have a strong opinion here, but just a suggestion.